jt-classbreaks/src/main/java/it/geosolutions/jaiext/classbreaks/QuantileBreaksOpImage.java

```diff
@@ -116,6 +117,19 @@ protected void postCalculate(Classification c, int band) {
                 set.add(e.getKey());
             }
         }
-        qc.setBreaks(band, set.toArray(new Double[set.size()]));
+        int actualClassNum = set.size();
+        qc.setBreaks(band, set.toArray(new Double[actualClassNum]));
+        if (this.percentages.booleanValue())
+            qc.setPercentages(getPercentages(dSize, actualClassNum));
+    }
+
+    private double[] getPercentages (double nvalues,int classNum){
+        double size = nvalues / (double) classNum;
+        double [] percentages = new double[classNum];
+        for (int i=0; i<classNum; i++){
+            percentages[i]= (size/nvalues)*100;
```

Comment:
Quantile tries to build equal classes, data might disagree and make it impossible, please add a test with a raster having only 3 values, 5 classes, and one of the values taking half of the pixels, while the other 2 are getting 25% each. Check that the result is correct and percentages are too.

Reply:
fixed